### DISCUSSION TITLE
Trying to setup second Sharepoint

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#2231] [Q&A] [answered]
Hi there, as you will note, I just had a scare with the [GUI](https://github.com/abraunegg/onedrive/discussions/2230) but this seems to have been fixed.

Meanwhile, I want to attach a second Sharepoint site to my onedrive. I've setup the config files, got it authenticated, but when try to get the drive error:

```
john@mint:~/.config/sharepoint_usfoffice$ onedrive --get-O365-drive-id 'my_search_query_for_drive'
Configuration file successfully loaded

WARNING: Your onedrive client version is obsolete and unsupported. Please upgrade your client version.

Configuring Global Azure AD Endpoints

ERROR: onedrive application is already running - check system process list for active application instances

Segmentation fault (core dumped)
```

Any suggestions?

n.b. It says my version is obsolete... but I installed via package so not sure why it doesn't update by default. I am on 2.4.18

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
> n.b. It says my version is obsolete... but I installed via package so not sure why it doesn't update by default. I am on 2.4.18

Are you using Ubuntu ???

This is the reason why. 

If you are using Ubuntu (you most likely are as Ubuntu is the *only* distribution with an old version - https://repology.org/project/onedrive/versions), your options are:

* Install the client on Ubuntu the *correct* way as per: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md
* Become the package maintainer for Ubuntu
* Dont use use Ubuntu and use a better maintained distribution

**Comment by johnsidoli** [comment]
That is the same way I installed it I think? I followed these instructions carefully and did it this way.

Meanwhile, I fixed the seg fault above by prepending sudo as per this discussion:

https://github.com/abraunegg/onedrive/discussions/2070

p.s. when I try to update my system I get this error at the end... is this related to why onedrive isn't updating? n.b. I absolutely [THINK THAT I] followed the instructions on here (github) [BUT I MAY BE WRONG!!] when I installed this way back when...

```
Err:11 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04 ./ InRelease
  The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
Fetched 355 kB in 1s (266 kB/s)
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04 ./ InRelease: The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
W: Failed to fetch https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04/./InRelease  The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

**Comment by abraunegg** [comment]
@johnsidoli 
https://github.com/abraunegg/onedrive/discussions/2070 clearly states *do not use sudo* ....

![image](https://user-images.githubusercontent.com/4956234/202922669-7e013a8b-fd03-4556-ab10-92456218e621.png)

Do Not Use 'sudo'

Regarding this error:
> ```
> Err:11 https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04 ./ InRelease
>   The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project > <home:npreining@build.opensuse.org>
> Fetched 355 kB in 1s (266 kB/s)
> Reading package lists... Done
> W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04 ./ InRelease: The following > signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project <home:npreining@build.opensuse.org>
> W: Failed to fetch https://download.opensuse.org/repositories/home:/npreining:/debian-ubuntu-onedrive/xUbuntu_20.04/./InRelease  > The following signatures were invalid: EXPKEYSIG B8AC39B0876D807E home:npreining OBS Project > <home:npreining@build.opensuse.org>
> W: Some index files failed to download. They have been ignored, or old ones used instead.
> ```

This is again because of Ubuntu.

You need to probably *remove* the repository and then *re-add* the OpenSuSE repository ..... this is all because of your choice of operating system .... 

Unfortunately I cannot assist with that problem - speak to folk who support Ubuntu to assist there.

**Comment by johnsidoli** [comment]
Ah, balls, OK, I'll look into this. Meanwhile thanks @abraunegg for the great advice as ever!

n.b. Am using Linux Mint (ubuntu of course)

**Reply by abraunegg** [reply]
@johnsidoli 
> n.b. Am using Linux Mint (ubuntu of course)

Best then post a question on https://forums.linuxmint.com/ regarding the `An error occurred during the signature verification` and ask how to fix & how to automatically keep the repo & signature up-to-date.

**Comment by johnsidoli** [comment]
p.s. the alternative suggestion of re-authorising didn't work, but I have my drive ID now from last time when I used sudo...
